src/poll.rs

```diff
@@ -1161,7 +1161,7 @@ impl Poll {
         self.readiness_queue.poll(&mut events.inner);
 
         // Return number of polled events
-        Ok(events.len())
+        Ok(events.iter().count())
```

Comment:
What about a 0 or 1?  Return 0 for no events (like a timeout) and 1 if there is at least one?